src/lib/libast/include/ast_sys.h

```diff
@@ -122,8 +122,10 @@ extern int	spawnvex_add(Spawnvex_t*, intmax_t, intmax_t, Spawnvex_f, void*);
 extern int	spawnvex_apply(Spawnvex_t*, int, int);
 extern intmax_t	spawnvex_get(Spawnvex_t*, int, int);
 extern int	spawnvex_close(Spawnvex_t*);
+#if !defined(__DARWIN_C_LEVEL) || !defined(__clang__)
```

Comment:
@catull, Can you explain this fix? It doesn't make sense to me. First, macOS 10.12 provides these two functions and defines `size_t`. And the declarations inside this conditional block are essentially the same as those provided by the system headers. The compiler diagnostic implies that `size_t` is either not defined or incorrectly defined.

Reply:
In short, because the functions are macro-expanded to
```c
extern size_t __builtin___strlcat_chk (char*, const char*, size_t, __builtin_object_size (char*, 2 > 1 ? 1 : 0));
extern size_t __builtin___strlcpy_chk (char*, const char*, size_t, __builtin_object_size (char*, 2 > 1 ? 1 : 0));
```

IMHO `size_t` is defined and is not the culprit here.
I have a few questions to you.

1. Can you compile the sources on macOS 10.13.x out of master ?
2. Why are those functions redefined if they "are essentially the same as those provided by the system headers" ?  What is the motivation ?

I have prepared a new patch, which also indents the preprocessor directives, because it is hard to follow the nesting of `#ifdef` / `#else` / `#endif`.
In the past, you could not have white-spaces before a `#` and after it, either.
Modern C compiler have supported it for a while now.
Not suggesting this is a style we should adopt, just saying the code is easier to follow.